Tests/Editor/CoreApi/ManagedEC2DeploymentTests.cs

```diff
@@ -0,0 +1,201 @@
+        
+        [SetUp]
+        public void Setup()
+        {
+            _awsCredentialsTestProvider = new AwsCredentialsTestProvider();
+            _managedEc2FleetParametersMock = new Mock<ManagedEC2FleetParameters>();
+            _coreApiMock = new Mock<CoreApi>();
+        }
+
+        private ManagedEC2Deployment ArrangeEc2DeploymentHappyPath()
+        {
+            _coreApiMock.Setup(f => f.PutSetting(It.IsAny<SettingsKeys>(), It.IsAny<string>())).Returns(Response.Ok(new PutSettingResponse()));
+            _coreApiMock.Setup(f => f.PutSettingOrClear(It.IsAny<SettingsKeys>(), It.IsAny<string>())).Returns(Response.Ok(new PutSettingResponse()));
+            _coreApiMock.Setup(f => f.PutSetting(It.IsAny<SettingsKeys>(), null)).Returns(Response.Fail(new PutSettingResponse()));
+            _coreApiMock.Setup(f => f.PutSetting(It.IsAny<SettingsKeys>(), string.Empty)).Returns(Response.Fail(new PutSettingResponse()));
+            _coreApiMock.Setup(f => f.GetSetting(It.IsAny<SettingsKeys>())).Returns(Response.Ok(new GetSettingResponse()));
+
+            _managedEc2FleetParametersMock.Object.GameName = GameName;
+            _managedEc2FleetParametersMock.Object.FleetName = FleetName;
+            _managedEc2FleetParametersMock.Object.BuildName = BuildName;
+            _managedEc2FleetParametersMock.Object.GameServerFile = GameServerFile;
+            _managedEc2FleetParametersMock.Object.GameServerFolder = GameServerFolder;
+            _managedEc2FleetParametersMock.Object.OperatingSystem = OperatingSystemName;
+            
+            const string testScenarioName = "test scenario";
+            const string testScenarioDescription = "test scenario description";
+            const string testScenarioUrl = "test";
+
+            Mock<ScenarioLocator> scenarioLocatorMock = SetUpScenarioLocatorToReturnTestDeployer(
+                testScenarioName, testScenarioDescription, testScenarioUrl, hasServer: false, _coreApiMock);
+
+            _deploymentSettings = GetUnitUnderTest(scenarioLocator: scenarioLocatorMock);
+            
+            _deploymentSettings.Refresh();
+            return new ManagedEC2Deployment(_deploymentSettings, _managedEc2FleetParametersMock.Object);
+        }
+        
+        [Test]
+        public void StartDeployment_WhenCorrectInputs_ExpectSuccess()
+        {
+            //Arrange
+            var ec2DeploymentHappyPath = ArrangeEc2DeploymentHappyPath();
+            
+            //Act
+            ec2DeploymentHappyPath.StartDeployment();
+            
+            //Assert
+            _coreApiMock.Verify(f => f.PutSetting(It.IsAny<SettingsKeys>(), It.IsAny<string>()), Times.Exactly(8));
+            
+            Assert.AreEqual(_deploymentSettings.GameName, GameName);
+        }
+        
+        [Test]
+        public void StartDeployment_WhenTaskIsFaulted_ExpectException()
+        {
+            //Arrange
+            var ec2DeploymentHappyPath = ArrangeEc2DeploymentHappyPath();
```

Comment:
Why are we creating this happy path resource that we don't use?

Reply:
I think I answered my own question which is that we're using the constructor of ManagedEC2Deployment for testing DeploymentSettings which inherently doesn't make sense for a unit test

Reply:
Removed File.